src/v/cluster/log_eviction_stm.cc

```diff
@@ -43,14 +43,59 @@ log_eviction_stm::log_eviction_stm(
   : base_t("log_eviction_stm.snapshot", logger, raft, kvstore) {}
 
 ss::future<> log_eviction_stm::start() {
-    ssx::spawn_with_gate(_gate, [this] { return monitor_log_eviction(); });
-    ssx::spawn_with_gate(
-      _gate, [this] { return handle_log_eviction_events(); });
+    setup_metrics();
+    // setup state watchdog timer
+    _loop_state.state_watchdog_timer.set_callback([this] {
+        auto holder = _gate.hold();
```

Comment:
unclear to me if its ok for the timer call back to throw, may be it is (not clear from quick scan of the code).